tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -39,6 +45,96 @@ private void getCapabilities(PrintWriter out, HttpServletRequest hsreq) {
 		gcdw.finishXML();
 	}
 	
+	/**
+	 * Checks request parameters for errors.
+	 * Will send back an XML Exception if any errors are encountered.
+	 * 
+	 * @param request parameter value
+	 * @param version parameter value
+	 * @param service parameter value
+	 * @return an ExceptionWriter if any errors occurred or null if none occurred
+	 */
+	private WFSExceptionWriter checkParametersForError(String request, String version, String service) {
+		// The SERVICE parameter is required. If not specified, is an error (throw exception through XML).
+		if(service != null) {
+			// For the WFS servlet it must be WFS if not, write out an InvalidParameterValue exception.
+			if(!service.equalsIgnoreCase("WFS")) {
+					return new WFSExceptionWriter("WFS Server error. SERVICE parameter must be of value WFS.", "service", "InvalidParameterValue");
+			}
+			
+		}
+					
+		else {
+			return new WFSExceptionWriter("WFS server error. SERVICE parameter is required.", "request", "MissingParameterValue");
+		}
+		
+		// The REQUEST Parameter is required. If not specified, is an error (throw exception through XML).
+		if(request != null) {
+			
+			// Only go through version checks if NOT a Get Capabilities request, the VERSION parameter is required for all operations EXCEPT GetCapabilities section 7.6.25 of WFS 2.0 Interface Standard

```

Comment:
Cool! I didn't know that `version` was optional with `GetCapabilities` requests. Good spec-checking!